### PR TITLE
fix: posting results from tests via Actions

### PR DESCRIPTION
**Pull request by mihai-sysbio**
### Main improvements in this PR:

This PR fixes the parsing of the tests result that are run with GH Actions, thus improving the short version of the results that gets posted automatically to the PR thread.

**I hereby confirm that I have:**

- [ ] Tested my code on my own machine
- [x] Followed the [development guidelines](https://github.com/SysBioChalmers/RAVEN/wiki/DevGuidelines).
- [x] Selected `devel` as a target branch
- [x] If needed, asked first in the [Gitter chat room](https://gitter.im/SysBioChalmers/RAVEN) about this PR

### CODE REVIEW

**Comment by edkerk** [comment]
It seems like the MATLAB output is still cut after a certain length? Looking at the edit history of the github-actions bot [post](https://github.com/SysBioChalmers/RAVEN/pull/411#issuecomment-1106073698), it seems like the post from 22 April 2022 10:11 CEST showed the longer text?

Also, I'm not sure how this exactly works, but would it be possible to keep the line-breaks in the MATLAB output? Considering it is a markdown document, maybe each sentence should end with two spaces?

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -7,22 +7,21 @@ jobs:
...
+      - name: Checkout
+        uses: actions/checkout@v2
 
-    - name: Checkout
-      uses: actions/checkout@v2
+      - name: Run tests
+        id: matlab-test
+        run: |
+          TEST_RESULTS=$(/usr/local/bin/matlab -nodisplay -nosplash -nodesktop -r "addpath(genpath('.')); cd('testing/unit_tests'); runtests(struct2table(dir('*.m')).name); exit;")
+          PARSED_RESULTS=$(echo $TEST_RESULTS | awk -F'com\.' '{ n = split($2, v, "(___+)|(===+)|(---+)"); for (i = 0; ++i <= n;) { print v[i] } }')
```
Comment (mihai-sysbio) [inline .github/workflows/testing.yml:17]: This is the core part of the fix:
```suggestion
          PARSED_RESULTS=$(echo $TEST_RESULTS | awk -F'com\\.' '{ n = split($2, v, "(___+)|(===+)|(---+)"); for (i = 0; ++i <= n;) { print v[i] } }')
```

**Comment by mihai-sysbio** [comment]
> It seems like the MATLAB output is still cut after a certain length? Looking at the edit history of the github-actions bot [post](https://github.com/SysBioChalmers/RAVEN/pull/411#issuecomment-1106073698), it seems like the post from 22 April 2022 10:11 CEST showed the longer text?

Right! This is resolved in 867189a79d869c0f5720fc03f844ae91e8e7ab51.

> Also, I'm not sure how this exactly works, but would it be possible to keep the line-breaks in the MATLAB output? Considering it is a markdown document, maybe each sentence should end with two spaces?

I've tried, but the text is shared via a bash variable and at some point the newline gets removed.

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -7,22 +7,21 @@ jobs:
...
+        uses: actions/checkout@v2
 
-    - name: Checkout
-      uses: actions/checkout@v2
+      - name: Run tests
+        id: matlab-test
+        run: |
+          TEST_RESULTS=$(/usr/local/bin/matlab -nodisplay -nosplash -nodesktop -r "addpath(genpath('.')); cd('testing/unit_tests'); runtests(struct2table(dir('*.m')).name); exit;")
+          PARSED_RESULTS=$(echo $TEST_RESULTS | awk -F'com\\.' '{ n = split($2, v, "(___+)|(===+)|(---+)"); for (i = 0; ++i <= n;) { print v[i] } }')
+          echo ::set-output name=results::$PARSED_RESULTS
```
Comment (edkerk) [inline .github/workflows/testing.yml:18]:
```suggestion
          PARSED_RESULTS=$(echo "$TEST_RESULTS" | awk -F'com\\.' '{ n = split($2, v, "(___+)|(===+)|(---+)"); for (i = 0; ++i <= n;) { print v[i] } }')
          echo ::set-output name=results::"$PARSED_RESULTS"
```

Might this resolve the issue with the line breaks, as suggested [here](https://stackoverflow.com/questions/22101778/how-to-preserve-line-breaks-when-storing-command-output-to-a-variable)?

**Review by edkerk — CHANGES_REQUESTED** [review]

File: .github/workflows/testing.yml (hunk 2 above)
Comment (edkerk) [reply .github/workflows/testing.yml:18]: This should have been a "comment", not "request changes".

Comment (mihai-sysbio) [reply .github/workflows/testing.yml]: Thanks for the input - it seemed the problem is how GH Actions is dealing with some chars.

**Pull request merged**